cpp/ql/src/Diagnostics/FailedExtractions.ql

```diff
@@ -6,17 +6,11 @@
  */
 
 import cpp
+import FailedExtractions
 
-class AnonymousCompilation extends Compilation {
-  override string toString() { result = "<compilation>" }
-}
-
-string describe(Compilation c) {
-  if c.getArgument(1) = "--mimic"
-  then result = "compiler invocation " + concat(int i | i > 1 | c.getArgument(i), " " order by i)
-  else result = "extractor invocation " + concat(int i | | c.getArgument(i), " " order by i)
-}
-
-from Compilation c
-where not c.normalTermination()
-select c, "Extraction failed for " + describe(c), 2
+from ExtractionError error
+where
+  error instanceof ExtractionUnknownError or
+  exists(error.getFile().getRelativePath())
+select error, "Extracting file $@ failed with $@ (at $@)", error.getFile(), error.getErrorMessage(),
+  error.getLocation(), error.getSeverity()
```

Comment:
The "placeholder pairs" are where we want a `<location, string>` pair split into 2 columns.

Here we only have a string, so we should just concatenate it, i.e.
```
select error, "Extracting file " + error.getFile() + " failed with " + error.getErrorMessage() + " (at " + error.getLocation().toString() + ")", error.getSeverity()
```